File: WebApi/Api/HttpClasses/Output/RoomJoinOutput.cs
using WebApi.Api.DTO;
using WebApi.Core.Entities;
using WebApi.Core.Entities.In_memory;

namespace WebApi.Api.HttpClasses.Output;

public class RoomJoinOutput
{
    public string AuthorizationToken { get; set; }
    public bool IsAdmin { get; set; }

    public List<ChatMessage> ChatMessages { get; set; }
    public List<PlaylistVideo> PlaylistVideos { get; set; }
    public List<UserDTO> Users { get; set; }
    public RoomSettings RoomSettings { get; set; }
    public UserPermissions UserPermissions { get; set; }
    public VideoPlayer VideoPlayer { get; set; }
}

File: ReactApp/src/types/HttpTypes/Output/RoomJoinOutput.ts
import { ChatMessage } from "../../ChatMessage";
import { PlaylistVideo } from "../../PlaylistVideo";
import { UserPermissions } from "../../UserPermissions";
import { User } from "../../User";
import { VideoPlayer } from "../../VideoPlayer";
import { RoomSettings } from "../../RoomSettings";

export interface RoomJoinOutput {
  //User fields
  authorizationToken: string;
  isAdmin: boolean;
  //Room fields
  chatMessages: ChatMessage[];
  playlistVideos: PlaylistVideo[];
  users: User[];
  roomSettings: RoomSettings;
  userPermissions: UserPermissions;
  videoPlayer: VideoPlayer;
}
